### ISSUE TITLE
Build issue on IOS

### ISSUE THREAD

**Issue by jp928** [airbnb/lottie-react-native#175]
react-native: 0.47.1
platform: ios
node: 8.4.0
lottie-react-native version is 2.2.0

installation method:

react-native link lottie-ios
react-native link lottie-react-native

issue: 
cannot build even after following the entire installation instructions.

<img width="894" alt="screen shot 2017-08-18 at 10 25 49 am" src="https://user-images.githubusercontent.com/659612/29439250-de6692fc-83ff-11e7-9c95-503b3a74027a.png">

**Comment by jp928** [comment]
BTW, lottie-react-native version is 2.2.0

**Comment by jarrrgh** [comment]
Hmm hmm, seems like 'react-native unlink lottie-ios' solved this for me - not quite sure why. I used npm instead of pods, if that makes a difference.

**Comment by micksabox** [comment]
I'm facing a similar issue. I tried ensuring `Header Search Paths` is point correctly to the header files. My `lottie-ios` version is `2.0.5`

**Comment by jarrrgh** [comment]
I'm not sure what I actually did to fix it, but I think I did'react-native unlink lottie-ios' 'react-native link lottie-ios'

**Comment by micksabox** [comment]
@jarrrgh Thanks, I ended up trying the same thing and it worked.

**Comment by edencakir** [comment]
@jp928 any updates?

**Comment by j-wang** [comment]
I tried @jarrrgh's solution, and unfortunately it did not work for me. I even tried clearing the cache, unlinking everything, relinking everything, cleaning the build, etc. No joy. :(

**Comment by j-wang** [comment]
Ok, finally got it working for me by:

- Removing node_modules (`rm -rf node_modules && npm install`)
- Clearing the packager cache (`rm -fr $TMPDIR/react-*`)
- react-native unlink lottie-ios
- Cleaning from XCode
- Clean Build Folder from XCode (it appears where Clean is, just hold option)
- react-native link lottie-ios

**Comment by micksabox** [comment]
This kept biting me so I looked into it further. Looks like there are [3 different ways](https://stackoverflow.com/questions/7439192/xcode-copy-headers-public-vs-private-vs-project) a header can be included into a project.

What I did was toggled the problematic files `LOTAnimationView.h` and `LOTAnimationTransitionController.h`

<img width="1393" alt="screen shot 2017-09-20 at 6 00 36 pm" src="https://user-images.githubusercontent.com/2997810/30669815-c0c083c8-9e2d-11e7-9044-d7aa2289041c.png">

I first went toggled them from `Public` to `Project`, tried building but it still failed. Then, I toggled it back to `Public` and it worked after a `Product > Clean`.

Hopefully this provides a lead for a further fix.

**Comment by tutts** [comment]
If you have to [add Lottie as an embedded binary](https://github.com/airbnb/lottie-react-native#troubleshooting), I had to remove the embedded binary, clean and then add the binary back in again for it to build :)

![image](https://user-images.githubusercontent.com/7454558/30693274-b75c6880-9ec6-11e7-8a69-2abde4756930.png)

**Comment by micksabox** [comment]
I also experience when I `Edit Scheme` to a different build environment. My fix above only works if you sequence it like this:
- Toggle `Public` to `Project`
- Attempt build, expect failure
- Toggle back from `Project` to `Public`
- Attempt build, it works

**Comment by j-wang** [comment]
It seems with all of these solutions that there's something with the build caching that causes the problem. 

I don't expect to run into this again, given that I've now upgraded my project to 2.2.0 and any fresh git clone won't have these previous build caches... but I do wonder what it is that causes Lottie to specifically have this problem, vs. all the other packages I've upgraded. 

I spent some time trying to track it down, but couldn't find the root cause and just spent a lot of time staring at XCode giving me errors that should have been immediately solved by the imported headers (and did if I just created a fresh header or dumped the interfaces directly into the .m file). That being said, I don't know the ins-and-outs of XCode's build system.
